clients/drcachesim/tracer/raw2trace.h

```diff
@@ -70,19 +71,196 @@ struct module_t {
     bool is_external; // If true, the data is embedded in drmodtrack custom fields.
 };
 
+const uint MAX_COMBINED_ENTRIES = 64;
+
 /**
- * The raw2trace class converts the raw offline trace format to the format
- * expected by analysis tools.  It requires access to the binary files for the
- * libraries and executable that were present during tracing.
+ * Header of raw trace.
+ */
+struct trace_header_t {
+    pid_t pid;
+    pid_t tid;
+    uint64 timestamp;
+};
+
+/**
+ * instr_summary is a compact encapsulation of the information needed by trace conversion
+ * from decoded instructions.
```

Comment:
This patch is very large, making it hard to find time to look at it...it would be easier and quicker to review if structured as a series of separate PR's: first, templatize the raw2trace class with zero functional changes, so that the diff with a lot of code movement doesn't also have code changes.  Add features like this instr data structure and the module mapper separately.  Smaller PR's, quicker reviews, smaller history changes in the repository, better future bisects: many benefits.  Would it be possible to split it up?

Reply:
I think it might be easier if the module mapper went first, I think it will avoid fields moving around in the template PL, making the latter cleaner. Let me give it a try - I'll send a PR shortly.